app/scripts/decisions.directive.js

```diff
@@ -23,42 +23,35 @@ angular.module('dashboard')
             self.supporter = null;
             self.voting = null;
             self.isTooltips = $rootScope.isTooltips;
+            var mtgTopicSelected = StorageSrv.getKey(CONST.KEY.TOPIC);
+            var mtgItemSelected = StorageSrv.getKey(CONST.KEY.MEETING_ITEM);
 
             // FUNCTIONS
 
-            function getDecisions(topic) {
-                if (angular.isObject(topic)) {
+            function getDecisions(aMtg, aTopic) {
+                $log.debug("dbDecisions.getDecisions", arguments);
+                if (angular.isObject(aTopic) && angular.isObject(aMtg)) {
                     self.record = null;
                     self.supporter = null;
                     self.voting = null;
 
-                    AhjoMeetingSrv.getDecisions(topic.topicGuid).then(function (data) {
-                        $log.log("dbDecisions.getDecisions");
-
-                        if (angular.isObject(data) && angular.isObject(data.objects)) {
-                            if (angular.isArray(data.objects.record)) {
-                                self.record = data.objects.record;
-                            }
-                            if (angular.isArray(data.objects.supporter)) {
-                                self.supporter = data.objects.supporter;
-                            }
-                            if (angular.isArray(data.objects.voting)) {
-                                self.voting = data.objects.voting;
-                            }
-                        }
-                        else {
-                            $log.error("dbDecisions.getDecisions: invalid response");
-                        }
-                    }, function (/*error*/) {
-                        $log.error("dbDecisions.getDecisions");
+                    AhjoMeetingSrv.getDecisions(aMtg.meetingGuid, aTopic.topicGuid).then(function (resp) {
+                        $log.log("dbDecisions.getDecisions", resp);
+
+                        self.record = resp.record;
+                        self.supporter = resp.supporter;
+                        self.voting = resp.voting;
+                    }, function (error) {
+                        // $log.error("dbDecisions.getDecisions");
+                        Utils.processAhjoResponse(error);
```

Comment:
This needs to be updated in further PRs to handle the custom error object now passed by the ui service.